win32ss/gdi/eng/surface.c

```diff
@@ -558,6 +558,16 @@ EngModifySurface(
     return TRUE;
 }
 
+/*
+ * @implemented
+ */
+BOOL
+APIENTRY
+NtGdiEngDeleteSurface(
+    _In_ HSURF hsurf)
+{
+    return EngDeleteSurface(hsurf);
+}
```

Comment:
This doesn't look like a complete implementation. Whilst indeed the system call invokes `EngDeleteSurface` to free the GDI surface object, it un-secures the memory of that handle and decrements the reference count before that. As you can see in this image.
![Capture](https://user-images.githubusercontent.com/34916900/116926665-eab9a080-ac5a-11eb-923b-4846612b8d23.PNG)

Although `MmUnsecureVirtualMemory` is not yet implemented in the Memory manager, you've to do some research to figure out how critical is the un-securing part. But invalidating the handle as-is can lead to abnormal behaviour.

Reply:
Not needed. Superseded by @jimtabor comment. :slightly_smiling_face: